src/plots/cartesian/autorange.js

```diff
@@ -226,8 +226,13 @@ function makePadFn(fullLayout, ax, max) {
     var A = 0;
     var B = 0;
     if(!isLinked(fullLayout, ax._id)) {
-        A = padInsideLabelsOnAnchorAxis(ax, max);
-        B = padInsideLabelsOnThisAxis(ax, max);
+        if(ax.rangemode !== 'nonnegative') {
+            A = padInsideLabelsOnAnchorAxis(ax, max);
+
+            if(anchorAxis.rangemode !== 'nonnegative') {
```

Comment:
why would `rangemode` of `anchorAxis` be relevant?

Reply:
Good question. Fixed in https://github.com/plotly/plotly.js/pull/5331/commits/f31d97ce3741a6f8ae8832e0ea30ed5abcc11d84.

Reply:
Wait. I had to revert the commit.